frontend/javascripts/oxalis/model/data_connection_info.ts

```diff
@@ -1,31 +1,115 @@
-const ROUND_TRIP_TIME_SMOOTHER = 0.125;
-const BANDWIDTH_SMOOTHER = 0.125;
+import _ from "lodash";
+// const ROUND_TRIP_TIME_SMOOTHER = 0.125;
+// const BANDWIDTH_SMOOTHER = 0.125;
+
+import window from "libs/window";
+
+type CallableStatsUpdate = (stats: Stats) => void;
+type DataConnectionInfoEntry = {
+  endTime: number;
+  startTime: number;
+  loadedBytesInMb: number;
+};
+
+type Stats = {
+  avgRoundTripTime: number;
+  avgDownloadSpeedInMBperS: number;
+  accumulatedDownloadedBytes: number;
+};
+
+const defaultStats: Stats = {
+  avgRoundTripTime: 0,
+  avgDownloadSpeedInMBperS: 0,
+  accumulatedDownloadedBytes: 0,
+};
 
 class DataConnectionInfo {
-  roundTripTime: number;
-  bandwidth: number;
-  totalBuckets: number;
-  totalBytes: number;
+  data: DataConnectionInfoEntry[];
+  subscriber: Array<CallableStatsUpdate>;
+  lastStats: Stats;
+  accumulatedDownloadedBytes: number;
 
   constructor() {
     // Give some typical initial values here to allow selection of initial loading strategy
-    this.roundTripTime = 200;
-    this.bandwidth = 100000;
-    this.totalBuckets = 0;
-    this.totalBytes = 0;
+    this.data = [];
+    this.subscriber = [];
+    this.lastStats = defaultStats;
+    this.accumulatedDownloadedBytes = 0;
+    setInterval(this.informAboutStatsUpdate, 1500);
+  }
+
+  _informAboutStatsUpdate() {
+    // calculate stats update
+    // Pass stats to all subscribers
+  }
+
+  informAboutStatsUpdate = () => {
+    const stats = this.getStatistics();
+    if (_.isEqual(this.lastStats, stats)) {
+      return;
+    }
+    this.subscriber.forEach((notifyFunction) => notifyFunction(stats));
+    this.lastStats = stats;
+  };
+
+  onStatisticUpdates(notifyFunction: CallableStatsUpdate): () => void {
+    this.subscriber.push(notifyFunction);
+    const unsubscribe = () => {
+      this.subscriber = this.subscriber.filter(
+        (currentFunc: CallableStatsUpdate) => currentFunc !== notifyFunction,
+      );
+    };
+    return unsubscribe;
+  }
```

Comment:
I'd remove this listener logic here and instead do that on the caller side (i.e., the status bar should do this). In the react component, you can simply use `useInterval` to poll the newest stats. That should be less code and also has the advantage that the polling stops, when the component is unmounted (I know, you are currently unsubscribing but the interval itself is never cleared).

Reply:
I meant the hook `useInterval` which is defined in react_helpers.ts :) I hope this just works :tm: If not, please let me know!

Reply:
Sadly, the `setInterval` hook does not fix the problem. But this seems to be related to Chrome + MacOS. With firefox + MacOS the statusbar happily updates its download speed all the time without any user interaction .... :/

Reply:
> Sadly, the setInterval hook does not fix the problem. But this seems to be related to Chrome + MacOS. With firefox + MacOS the statusbar happily updates its download speed all the time without any user interaction .... :/

Looks like I made some mistakes while testing with Chrome yesterday. Today the issue did not occur anymore. Therefore this pr is ready for the 2nd review round 🔔